js/src/forum/components/FlagPostModal.js

```diff
@@ -77,7 +95,7 @@ export default class FlagPostModal extends Modal {
               className="Button Button--primary Button--block"
               type="submit"
               loading={this.loading}
-              disabled={!this.reason()}>
+              disabled={!this.reason() || this.reason() === 'other' && !this.reasonDetail()}>
```

Comment:
Let's not add this clause. Disabled buttons without an error message explaining why can confuse users.

Reply:
In conjunction with the language changes (https://github.com/flarum/lang-english/pull/148/commits/420db03571fd833d95acce512db2ab6e64947ccb#diff-19cc7a6cd74cd8da7215054764db7577L31), this field is marked as 'please specify', whereas the other fields are marked 'optional'

Excuse the giffgaff theme for this screenshot:
![image](https://user-images.githubusercontent.com/16573496/69423914-3f938080-0d1f-11ea-98dc-63a077f7cbb9.png)

![image](https://user-images.githubusercontent.com/16573496/69423987-65b92080-0d1f-11ea-9411-da03d5c7a9fa.png)

Our reasoning behind this, is that we see in excess of 40% of flags raised as 'other' with no other content. The button is re-enabled once at least 1 character is entered.

Reply:
I agree "Other" flags without text content should be prevented, but that should be made clear through an explicit error message.

If people gloss over that small help text, they won't suddenly notice it while wondering why they cannot click the button.

Reply:
Some more nuance:
- [This StackOverflow answer](https://ux.stackexchange.com/a/73733/17557) presents some great reasoning for this pattern, with nice examples how this can improve the experience. I can relate to that, but...
- On the other hand, especially in terms of accessibility, [disabled buttons suck](https://axesslab.com/disabled-buttons-suck/). While it is possible to make this work well for e.g. users of screenreaders, it would be considerably more work.

In summary, I would like to request a) an always-enabled submit button, and b) an error message in case "Flag Post" is clicked too early.

We could consider disabling the button visually in addition, to provide the nice experience described in the first link.